graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -187,24 +191,21 @@ abstract class BaseDataTypeGenerator(internal val packageName: String, config: C
         fieldSpecs.mapIndexed { index, fieldSpec ->
             when (val fieldSpecType = fieldSpec.type) {
                 is ParameterizedTypeName -> {
-                    when (fieldSpecType.typeArguments[0]) {
-                        ClassName.get(String::class.java) -> {
-                            addToStringForListOfStrings(fieldSpec, javaType)
+                    if (typeUtils.isStringInput(fieldSpecType.typeArguments[0])) {
+                            val name = "serializeListOf" + (fieldSpecType.typeArguments[0] as ClassName).simpleName()
```

Comment:
This cast doesn't always work, I found some corner cases working on the composed schema. Needs a type check.